Add instructions on how to secure graphql (#2643)
kyma-project/kyma: components/ui-api-layer/docs/security.md

The `@HasAccess` directive is used to secure the action or a field in a type. It is used as a middleware before the resolver code is executed. 

Use the following query as an example on how to secure a GraphQL action:
```
limitRanges(namespace: String!): [LimitRange!]! @HasAccess(attributes: {resource: "limitranges", verb: "list", apiGroup: "", apiVersion: "v1", namespaceArg: "Namespace", isChildResolver: false})
```

Reference this table for details on the elements that make up a defined and secured GraphQL action:

| Defined GraphQL action element | Description |
|:----------|:------|
| `limitRanges(namespace: String!)` |  Name of the action, which in this case is a query, followed by a string that specifies the Namespace of the queried resources. |
| `: [LimitRange!]!` | Defines the type of object returned by the query. In this case, it's a list of LimitRanges resources. |
| `@HasAccess(attributes:` | Defines the GraphQL directive that secures access to the resource. |
| `resource: "limitranges"` | Defines the type of secured Kubernetes resource, in this case all `limitranges` resources. |

> | `verb: "list"` | Defines the secured interaction type. It is related to action type, which in this case is "query". |

Reviewer: Add some guidelines on what verbs should be used in which scenario. You may utilise what is already in security docs.

Author: done

| `apiGroup: ""` | Defines the apiGroup to which the user must have access to get the result of this query. In this case it is empty because limitRanges is the resource built into Kubernetes, not some Custom Resource created by us. |
| `apiVersion: "v1alpha1"` | Specifies the apiVersion of the query subject. |
| `namespaceArg: "Namespace"` | Specifies the name of the argument or field in the parent object from which the resource Namespace is fetched. |

> | `isChildResolver: false` | (Optional) Specifies if the directive is set on a field in a type. Determines if the Namespace should be fetched from the parent object instead of the argumentsnot available in the child object. By default it is set to `false`. |

Reviewer: ```suggestion
| `isChildResolver: false` | (Optional) Specifies if the directive is set on a field in a type. Determines if the Namespace should be fetched from the parent object instead of the arguments not available in the child object. By default it is set to `false`. |
```

Author: no longer relevant

| `: [LimitRange!]!` | Defines the type of object returned by the query. In this case, it's a list of LimitRanges resources. |
| `@HasAccess(attributes:` | Defines the GraphQL directive that secures access to the resource. |
| `resource: "limitranges"` | Defines the type of secured Kubernetes resource, in this case all `limitranges` resources. |
| `verb: "list"` | Defines the secured interaction type. It is related to action type, which in this case is "query". Use the the table from [this paragraph](https://kyma-project.io/docs/master/components/security#details-graphql) to choose the right verb. |
| `apiGroup: ""` | Defines the apiGroup to which the user must have access to get the result of this query. In this case it is empty because limitRanges is the resource built into Kubernetes, not some Custom Resource created by us. |
| `apiVersion: "v1alpha1"` | Specifies the apiVersion of the query subject. |
| `namespaceArg: "namespace"` | Specifies the name of the argument or field in the parent object from which the resource namespace is fetched. |

> | `isChildResolver: false` | Required to be "true" for fields nested in types which have to be secured. Determines if the namespace argument should be fetched from the parent object. By default it is set to `false`. |

Reviewer 2: ```suggestion
| `isChildResolver: false` | Must be "true" for fields nested in types which have to be secured. Determines if the `namespace` argument should be fetched from the parent object. By default, it is set to `false`. |
```

> If the directive is set on a field nested in a type, the value returned from a query returning that type depends on the type modifier set on the field. If the field is set as Non-Null, as described [here](https://graphql.github.io/learn/schema/#lists-and-non-null), the error will be returned in case the user has no rights to access the child resource. If the field is not set as Non-Null, only the child field will be returned as null.

Reviewer 2: Is "type" the name of a modifier? 
If yes, format it as `code`

Author: Type is not the name of modifier. `Non-Null` is. I will format it.

> If the directive is set on a field nested in a type, the value returned from a query returning that type depends on the type modifier set on the field. If the field is set as Non-Null, as described [here](https://graphql.github.io/learn/schema/#lists-and-non-null), the error will be returned in case the user has no rights to access the child resource. If the field is not set as Non-Null, only the child field will be returned as null.

Reviewer 2: ```suggestion
If the directive is set on a field nested in a type, the value returned from a query returning that type depends on the type modifier set on the field. If the field is set as [Non-Null](https://graphql.github.io/learn/schema/#lists-and-non-null), the error is returned for users who do have rights to access the child resource. If the field is not set as Non-Null, only the child field will be returned as null.
```

Author: Suggestion applied.
